lib/validators/custom-styles-validator.ts

```diff
@@ -0,0 +1,33 @@
+import { ComponentSet } from '../models';
+import { Validator } from './validator';
+import * as path from 'path';
+
+export class CustomStylesValidator extends Validator {
+    constructor(error: (errorMessage: string) => false, definition: ComponentSet, private filePaths: Set<string>) {
+        super(error, definition);
+    }
+
+    async validate(): Promise<void> {
+        this.checkCustomStylesFolder();
+
+        if (!this.componentSet.customStyles) {
+            return;
+        }
+
+        this.componentSet.customStyles.forEach((customStyle) => {
+            if (!customStyle.default) {
+                return;
+            }
+
+            if (!this.filePaths.has(path.normalize(customStyle.default))) {
+                this.error(`The default file for custom style "${customStyle.label}" does not exist`);
+            }
+        });
+    }
+
+    private checkCustomStylesFolder() {
+        if (Array.from(this.filePaths).some((file) => file.toLowerCase().startsWith('styles/customstyles/'))) {
```

Comment:
So the customer can't provide a default config?

Reply:
They can. The default property should be a file path to any file in the component set. For example, with Apple News the default property will point to the `custom/apple-news/styles.json` file. This means we can already upgrade the component set. Once the UI is done you should be able to update the styles then. 

The backend however stores the custom styles within the styles folder. So when the component set already contains a styles/customStyles folder it will be very hard to merge and clean the directory. For now I decided that this folder just can't be used in the component set. The default styles can be stored anywhere else. Including in a customStyles folder in the root of the component set.